server/src/hicstat.ts

```diff
@@ -112,8 +112,127 @@ export async function do_hicstat(file: string, isurl: boolean): Promise<HicstatR
 		})
 	//console.log('Reading matrix ...')
 	out_data.normalization = normalization
+
+	// Fetch observed and expected values for all resolutions
+	const resolutions = out_data['Base pair-delimited resolutions'].concat(out_data['Fragment-delimited resolutions'])
+
+	for (const resolution of resolutions) {
+		const observedExpectedData: Record<string, ResolutionData> = {}
+
+		for (const chr1 of out_data.chrorder) {
+			for (const chr2 of out_data.chrorder) {
+				const observed = await getObservedValues(file, resolution, chr1, chr2)
+				const expected = await getExpectedValues(file, resolution, chr1, chr2)
+				const observedOverExpected = calculateObservedOverExpected(observed, expected)
+
+				// Store the data in resolutionsData
+				observedExpectedData[`${chr1}-${chr2}`] = {
+					observed,
+					expected,
+					observedOverExpected
```

Comment:
We may need to investigate this approach further. I'm curious if the calculations for the matrix type can be done in the client rather than loading them all at once. If we intend to add more functionality - like all the matrix types listed [the aiden's hic code](https://github.com/aidenlab/Juicebox/blob/master/src/juicebox/windowui/MatrixType.java#L109) - then we may load them asynchronously to avoid expensive processes. This is probably something we ought to discuss to Xin in case it's a moot point.